pkg/cdn/getter.go

```diff
@@ -14,6 +18,16 @@ package cdn
 type Getter interface {
 	// Get gets the base URL for the given module.
 	// Returns ErrNotFound if the module isn't found, and another
-	// non-nil error on any other error encountered
+	// non-nil error on any other error encountered.
+	//
+	// This method does not touch the CDN itself. Olympus should just
+	// redirect vgo to the CDN so that vgo downloads module data directly
+	// from there.
+	//
+	// If something went wrong getting the module metadata, returns a non-nil
+	// error.
 	Get(module string) (string, error)
+	// GetForVersion gets the module metadata about the given module/version.
+	// If something went wrong doing the get operation, returns a non-nil error.
+	GetForVersion(module, version string) (eventlog.Event, error)
```

Comment:
not really sure if this should be here. in CDN you have stored raw module, no info about eventlog, operations... maybe we should extend eventlog to provide such a functionality instead of CDN

Reply:
took offline